### ISSUE TITLE
Versioning doesn't work

### ISSUE THREAD

**Issue by hpatoio** [FriendsOfSymfony/FOSRestBundle#1443]
I'm on `Symfony 2.3` and I'm using version `1.8` of FOSRestBundle.

I've a custom view handler and I specify the version via custom header 

Significant part of `config.yml`

```
services:
    my_api.view_handler:
        parent: fos_rest.view_handler.default
        calls:
         - ['registerHandler', [ 'pkpass', ["@my_api.applewallet_handler", 'createResponse'] ] ]
         - ['registerHandler', [ 'ics', ["@my_api.icalendar_handler", 'createResponse'] ] ]

fos_rest:
[..]
    service:
        view_handler: my_api.view_handler
[..]
    versioning:
        enabled: true
        default_version: 3.0.0
        resolvers:
            custom_header:
                enabled: true
                header_name: X-My-Version
```

The problem is that the value of the version is not set for the context.

If I remove the custom view handler everything works. It seems like the problem is in the way `fos_rest.version_listener` is declared. The first parameters is `fos_rest.view_handler.default` service but if you setup a custom handler then the view handler is another one.

The standard definition is 

```
        <service id="fos_rest.version_listener" class="FOS\RestBundle\EventListener\VersionListener">
            <argument type="service" id="fos_rest.view_handler.default" />
            <argument type="service" id="fos_rest.versioning.chain_resolver" />
            <argument /> <!-- default media type version -->
            <tag name="kernel.event_listener" event="kernel.request" method="onKernelRequest" priority="33"  />
        </service>
```

changing it to 
```
        <service id="fos_rest.version_listener" class="FOS\RestBundle\EventListener\VersionListener">
            <argument type="service" id="fos_rest.view_handler" />
            <argument type="service" id="fos_rest.versioning.chain_resolver" />
            <argument /> <!-- default media type version -->
            <tag name="kernel.event_listener" event="kernel.request" method="onKernelRequest" priority="33"  />
        </service>
```
seems to fix the problem.

Am I missing something or the definition for `fos_rest.version_listener` is wrong ?

**Comment by Ener-Getick** [comment]
Indeed this should be changed :+1:

**Comment by Ener-Getick** [comment]
Fixed in #1445
